app/controllers/vm_common.rb

```diff
@@ -1677,6 +1677,38 @@ def set_form_vars
     session[:edit] = @edit
   end
 
+  def action_type(type, amount)
+    action_types = {
+      "advanced_settings"   => n_("Advanced Setting", "Advanced Settings", amount),
+      "disks"               => n_("Number of Disk", "Number of Disks", amount),
+      "drift_history"       => n_("Drift History", "Drift History", amount),
+      "event_logs"          => n_("Event Log", "Event Logs", amount),
+      "filesystem_drivers"  => n_("Filesystem Drivers", "Filesystem Drivers", amount),
+      "filesystems"         => n_("Filesystem", "Filesystems", amount),
+      "groups"              => n_("Group", "Groups", amount),
+      "guest_applications"  => n_("Guest Application", "Guest Applications", amount),
+      "hv_info"             => n_("Container", "Container", amount),
+      "kernel_drivers"      => n_("Kernel Driver", "Kernel Drivers", amount),
+      "linux_initprocesses" => n_("Init Process", "Init Processes", amount),
+      "os_info"             => n_("OS Info", "OS Info", amount),
+      "parent_vm"           => n_("Parent VM", "Parent VM", amount),
+      "patches"             => n_("Patch", "Patches", amount),
+      "processes"           => n_("Running Process", "Running Processes", amount),
+      "registry_items"      => n_("Registry Item", "Registry Items", amount),
+      "resources_info"      => n_("Resource", "Resources", amount),
+      "scan_history"        => n_("Scan History", "Scan History", amount),
+      "snapshot_info"       => n_("Snapshot", "Snapshots", amount),
+      "users"               => n_("User", "Users", amount),
+      "vmtree_info"         => n_("Genealogy", "Genealogy", amount),
+      "win32_services"      => n_("Win32 Service", "Win32 Services", amount)
+    }
```

Comment:
`ACTION_TYPES = { "advanced_settings"   => (amount) -> { n_("Advanced Setting", "Advanced Settings", amount) }, .....`
and then  `ACTION_TYPES[type].call(amount)`

probably better than the `switch` base approach because you can still use `.key?`